src/main/webapp/app/shared/layouts/main/main.component.ts

```diff
@@ -38,6 +44,16 @@ export class JhiMainComponent implements OnInit {
 
     ngOnInit() {
         this.router.events.subscribe((event) => {
+            if (event instanceof NavigationStart) {
+                const newShowSkeleton = this.shouldShowSkeleton(event.url);
```

Comment:
why do we need `newShowSkeleton` here? Please document the rationale in the code